castai/resource_node_template.go

```diff
@@ -219,10 +221,20 @@ func resourceNodeTemplate() *schema.Resource {
 							Description: "Max Memory (Mib) per node.",
 						},
 						FieldNodeTemplateStorageOptimized: {
-							Type:        schema.TypeBool,
-							Optional:    true,
-							Default:     false,
-							Description: "Storage optimized instance constraint - will only pick storage optimized nodes if true",
+							Type:          schema.TypeBool,
+							Optional:      true,
+							Default:       false,
+							Deprecated:    "Use `storage_optimized_enum` instead. Attribute will be removed in the next major release.",
```

Comment:
but we should remove it in this release as discussed, right? So if customer upgrades and have set this flag, we return an error that they have to remove this value and set enum value instead

Reply:
After some consideration, deprecating + conflict is a better approach here.  

Users getting clear warning about deprecation and what to do. 

```text
╷
│ Warning: Argument is deprecated
│
│   with castai_node_template.default_by_castai,
│   on castai.tf line 199, in resource "castai_node_template" "default_by_castai":
│  199:     storage_optimized  = false
│
│ Use `storage_optimized_enum` instead. Attribute will be removed in the next major release.
```

Reply:
I just tried to remove the field and check the terraform changes. 

This is the only warning users get. There is no additional info. They need to go and look at documentation to understand what happened. 
```text
╷
│ Error: Unsupported argument
│
│   on castai.tf line 41, in resource "castai_node_template" "default_by_castai":
│   41:     storage_optimized = false
│
│ An argument named "storage_optimized" is not expected here.
```

Reply:
hmm, but if we still allow setting storage_optimized, then we have a problem with false value right? Because customer could ignore this warning and then get a side effect that their flag started not working

Reply:
Yes, that's why the updated warning with explicit warning.  

I believe this migration is the most user-friendly approach if we're going to support it with proper change logs

```text
╷
│ Warning: Argument is deprecated
│
│   with castai_node_template.default_by_castai,
│   on castai.tf line 41, in resource "castai_node_template" "default_by_castai":
│   41:     storage_optimized = false
│
│ This attribute will be removed in the next major release. Use
│ `storage_optimized_status` instead. This field is currently only
│ processed if set to `true`.
│
│ (and one more similar warning elsewhere)
╵
```